### ISSUE TITLE
redundant_optional_initialization rule does not provide violations for ones inside the functions.

### ISSUE THREAD

**Issue by VivoEx** [realm/SwiftLint#2233]
We do not see violations with redundant_optional_initialization rule for the vars inside the fuctions neither does it get autocorrected whereas works ok over the class variables.
```javascript
class ClassName {
   ....
    var optionalVar: String? // This line provides violation and thus corrected during autocorrect
   
    func funcName() {
        var optionalVar: String? // This doesn't provide violation
        ....
    }

}
```
Swiftlint 0.25.1
Swift 4.1

**Comment by marcelofabri** [comment]
The problem here is that SourceKit doesn't return `key.setter_accessibility` for local variables. We use that to know if it's a `var` or a `let`, because if it's a `let`, we shouldn't warn.
